cloudformation/batch-substack.cfn.json

```diff
@@ -188,6 +188,144 @@
         ]
       }
     },
+    "BatchUserRole": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "MaxSessionDuration": 36000,
+        "AssumeRolePolicyDocument": {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": {
+                  "Fn::Sub": "arn:${AWS::Partition}:iam::${AWS::AccountId}:root"
+                }
+              },
+              "Action": "sts:AssumeRole",
+              "Condition": {}
+            }
+          ]
+        },
+        "Policies": [
+          {
+            "PolicyName": "BatchUserPolicy",
+            "PolicyDocument": {
+              "Version": "2012-10-17",
+              "Statement": [
+                {
+                  "Effect": "Allow",
+                  "Action": [
+                    "batch:SubmitJob",
+                    "batch:RegisterJobDefinition",
+                    "cloudformation:DescribeStacks",
+                    "ecs:ListContainerInstances",
+                    "ecs:DescribeContainerInstances",
+                    "logs:GetLogEvents",
+                    "logs:FilterLogEvents",
+                    "s3:PutObject",
+                    "s3:Get*",
+                    "s3:DeleteObject",
+                    "iam:PassRole"
+                  ],
+                  "Resource": [
+                    {
+                      "Fn::Sub": "${JobDefinitionSerial}"
+                    },
+                    {
+                      "Fn::Sub": [
+                        "${MNPJobDefinitionArn}*",
+                        {
+                          "MNPJobDefinitionArn": {
+                            "Fn::Select": [
+                              "0",
+                              {
+                                "Fn::Split": [
+                                  ":1END_OF_THE_STRING",
```

Comment:
Is this guaranteed to be revision `:1`?

Reply:
It should since we don't assign a static name to the resource, hence it's supposed not to conflict with an existing one.